app/src/org/commcare/connect/ConnectManager.java

```diff
@@ -0,0 +1,1034 @@
+        manager.parentActivity = parent;
+
+        if (manager.connectStatus == ConnectIdStatus.NotIntroduced) {
+            ConnectUserRecord user = ConnectUserDatabaseUtil.getUser(manager.parentActivity);
+            if (user != null) {
+                boolean registering = user.getRegistrationPhase() != ConnectConstants.CONNECT_NO_ACTIVITY;
+                manager.connectStatus = registering ? ConnectIdStatus.Registering : ConnectIdStatus.LoggedIn;
+
+                String remotePassphrase = ConnectDatabaseUtils.getConnectDbEncodedPassphrase(parent, false);
+                if (remotePassphrase == null) {
+                    getRemoteDbPassphrase(parent, user);
+                }
+            } else if (ConnectDatabaseHelper.isDbBroken()) {
+                //Corrupt DB, inform user to recover
+                ConnectDatabaseHelper.handleCorruptDb(parent);
+            }
+        }
+    }
+
+    public static void setPendingAction(int action) {
+        getInstance().pendingAction = action;
+    }
+
+    public static int getPendingAction() {
+        int action = getInstance().pendingAction;
+        getInstance().pendingAction = PENDING_ACTION_NONE;
+        return action;
+    }
+
+    public static BiometricManager getBiometricManager(CommCareActivity<?> parent){
+        ConnectManager instance = getInstance();
+        if (instance.biometricManager == null) {
+            instance.biometricManager = BiometricManager.from(parent);
+        }
+
+        return instance.biometricManager;
+    }
+
+    private static void scheduleHearbeat() {
+        if (AppManagerDeveloperPreferences.isConnectIdEnabled()) {
+            Constraints constraints = new Constraints.Builder()
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .setRequiresBatteryNotLow(true)
+                    .build();
+
+            PeriodicWorkRequest heartbeatRequest =
+                    new PeriodicWorkRequest.Builder(ConnectHeartbeatWorker.class,
+                            PERIODICITY_FOR_HEARTBEAT_IN_HOURS,
+                            TimeUnit.HOURS)
+                            .addTag(CONNECT_WORKER)
+                            .setConstraints(constraints)
+                            .setBackoffCriteria(
+                                    BackoffPolicy.EXPONENTIAL,
+                                    BACKOFF_DELAY_FOR_HEARTBEAT_RETRY,
+                                    TimeUnit.MILLISECONDS)
+                            .build();
+
+            WorkManager.getInstance(CommCareApplication.instance()).enqueueUniquePeriodicWork(
+                    CONNECT_HEARTBEAT_REQUEST_NAME,
+                    ExistingPeriodicWorkPolicy.REPLACE,
+                    heartbeatRequest
+            );
+        }
+    }
+
+    public static void setParent(Context parent) {
+        getInstance().parentActivity = parent;
+    }
+
+    public static boolean isConnectIdConfigured() {
```

Comment:
I would like this method to signify by name that it's checking for login as well. Maybe just call this `ConnectIdManager.isLoggedIn()`